apps/vpnmanager/src/lib/data/database.ts

```diff
@@ -0,0 +1,149 @@
+import betterSqlite3 from "better-sqlite3";
+import path from "path";
+
+const dbPath = path.resolve(process.cwd(), "data", "database.sqlite");
+const db = betterSqlite3(dbPath);
+
+class Record {
+  ID?: number;
+  userId: string;
+  usage: number;
+  date: string;
+  cumulativeData: number;
+  email: string;
+  accessUrl?: string;
+  createdAt: string;
+
+  constructor(
+    userId: string,
+    usage: number,
+    date: string,
+    cumulativeData: number,
+    email: string,
+    accessUrl?: string,
+    createdAt?: string,
+    ID?: number,
+  ) {
+    this.ID = ID;
+    this.userId = userId;
+    this.usage = usage;
+    this.date = date;
+    this.cumulativeData = cumulativeData;
+    this.email = email;
+    this.accessUrl = accessUrl;
+    this.createdAt = createdAt || new Date().toISOString();
+  }
+}
+
+export interface Filters {
+  email?: string;
+  date?: string;
+  dateBetween?: { start: string; end: string };
+  userId?: string;
+  ID?: number;
+  groupBy?: "email" | "date";
+  orderBy?: string;
+}
+
+class Model {
+  static initialize() {
+    const createTable = `
+      CREATE TABLE IF NOT EXISTS records (
+        ID INTEGER PRIMARY KEY AUTOINCREMENT,
+        userId TEXT NOT NULL,
+        usage INTEGER NOT NULL,
+        date TEXT NOT NULL,
+        cumulativeData INTEGER NOT NULL,
+        email TEXT NOT NULL,
+        accessUrl TEXT,
+        createdAt TEXT NOT NULL
+      )
+    `;
+    db.exec(createTable);
+  }
+
+  static create(record: Record) {
+    const insertData = db.prepare(`
+      INSERT INTO records (userId, usage, date, cumulativeData, email, accessUrl, createdAt)
+      VALUES (?, ?, ?, ?, ?, ?, ?)
+    `);
+    const info = insertData.run(
+      record.userId,
+      record.usage,
+      record.date,
+      record.cumulativeData,
+      record.email,
+      record.accessUrl,
+      record.createdAt,
+    );
+    return { ...record, ID: info.lastInsertRowid };
+  }
+
+  static update(ID: number, updates: Partial<Record>) {
+    const setClause = Object.keys(updates)
+      .map((key) => `${key} = ?`)
+      .join(", ");
+    const query = `UPDATE records SET ${setClause} WHERE ID = ?`;
+    const stmt = db.prepare(query);
+    return stmt.run([...Object.values(updates), ID]);
+  }
+
+  static delete(ID: number) {
+    const stmt = db.prepare("DELETE FROM records WHERE ID = ?");
+    return stmt.run(ID);
+  }
+
+  static getAll(filters: Filters = {}) {
+    let query = "SELECT";
+    const params: any[] = [];
+    if (filters.groupBy === "email" || filters.groupBy === "date") {
+      if (filters.groupBy === "email") {
+        query +=
+          " email, userId, accessUrl, SUM(usage) as totalUsage FROM records WHERE 1=1";
+      }
+      if (filters.groupBy === "date") {
```

Comment:
```suggestion
      } else {
```

At this point inside the main if, `groupBy` can only be `email` or `date`, no?